# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/esden/wvdial/gcc3_and_smallfix.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -Nur wvdial-1.42/Makefile wvdial-1.42.new/Makefile
--- wvdial-1.42/Makefile	Fri May 11 22:01:21 2001
+++ wvdial-1.42.new/Makefile	Tue Oct  1 01:07:02 2002
@@ -1,7 +1,7 @@
 TOPDIR=$(shell if [ "$$PWD" != "" ]; then echo $$PWD; else pwd; fi)
 
 DEBUG=0
-PREFIX=/usr/local
+PREFIX=/usr
 BINDIR=${PREFIX}/bin
 MANDIR=${PREFIX}/man
 PPPDIR=/etc/ppp/peers
@@ -20,7 +20,7 @@
 	[ -d ${MANDIR}/man1 ] || install -d ${MANDIR}/man1
 	[ -d ${PPPDIR}      ] || install -d ${PPPDIR}
 	install -m 0755 --strip src/wvdial/wvdial src/wvdial/wvdialconf ${BINDIR}
-	install -m 0644 wvdial.1 wvdialconf.1 ${MANDIR}/man1
+	install -m 0644 wvdial.1 wvdialconf.1 wvdial.conf.5 ${MANDIR}/man1
 	cp ppp.provider ${PPPDIR}/wvdial
 
 uninstall:
diff -Nur wvdial-1.42/src/streams/wvlog.h wvdial-1.42.new/src/streams/wvlog.h
--- wvdial-1.42/src/streams/wvlog.h	Wed Mar 15 21:44:53 2000
+++ wvdial-1.42.new/src/streams/wvlog.h	Tue Oct  1 00:52:21 2002
@@ -25,7 +25,7 @@
 // or transmits log messages.
 class WvLogRcvBase
 {
-    friend WvLog;
+    friend class WvLog;
 protected:
     const char *appname(const WvLog *log) const;
     virtual void log(const WvLog *source, int loglevel,
@@ -43,7 +43,7 @@
 // to all registered WvLogRcv's.
 class WvLog : public WvStream
 {
-    friend WvLogRcvBase;
+    friend class WvLogRcvBase;
 public:
     enum LogLevel {
 	Critical = 0,
diff -Nur wvdial-1.42/src/utils/wvstringtable.cc wvdial-1.42.new/src/utils/wvstringtable.cc
--- wvdial-1.42/src/utils/wvstringtable.cc	Sat Apr  1 12:40:47 2000
+++ wvdial-1.42.new/src/utils/wvstringtable.cc	Tue Oct  1 00:46:57 2002
@@ -8,7 +8,7 @@
 #include "strutils.h"
 
 
-WvString WvStringTable::join(const char *joinchars = " ")
+WvString WvStringTable::join(const char *joinchars)
 {
     WvStringTable::Iter s(*this);
     size_t totlen;
@@ -16,6 +16,8 @@
     char *te;
     int x;
     
+    if(joinchars == NULL) joinchars=" ";
+
     totlen = 1;
     for (s.rewind(); s.next(); )
 	totlen += strlen(s()) + strlen(joinchars);
diff -Nur wvdial-1.42/src/utils/wvtask.h wvdial-1.42.new/src/utils/wvtask.h
--- wvdial-1.42/src/utils/wvtask.h	Thu Mar 16 00:16:38 2000
+++ wvdial-1.42.new/src/utils/wvtask.h	Tue Oct  1 00:49:03 2002
@@ -21,7 +21,7 @@
 
 class WvTask
 {
-    friend WvTaskMan;
+    friend class WvTaskMan;
     typedef void TaskFunc(void *userdata);
     
     static int taskcount, numtasks, numrunning;
@@ -53,7 +53,7 @@
 
 class WvTaskMan
 {
-    friend WvTask;
+    friend class WvTask;
     WvTaskList free_tasks;
     
     void get_stack(WvTask &task, size_t size);
